build.rs

```diff
@@ -0,0 +1,37 @@
+extern crate pkg_config;
+
+use std::fs;
+use std::path::Path;
+use std::env;
+
+fn main() {
+    configure_snappy();
+}
+
+fn configure_snappy() {
+    // try pkg_config first
+    if pkg_config::find_library("snappy").is_ok() {
+        return;
+    }
+    if env::var_os("SNAPPY_STATIC").is_some() {
+        println!("cargo:rustc-link-lib=static=snappy");
+        println!("cargo:rustc-flags=-l c++");
+    } else {
+        println!("cargo:rustc-link-lib=dylib=snappy");
+    };
+
+    for f in vec!["/usr/lib","/usr/local/lib"] {
+        if is_file_in("libsnappy.a", Path::new(f)) {
+            println!("cargo:rustc-link-search={}", f);
+        }
+    }
```

Comment:
Oh yes, it's empty also on my systems by default. However, when installing libraries to non-standard paths (I have a user private `~/Library/opt/lib` for example) I use to set and export `LD_LIBRARY_PATH` in my `~/.profile` configuration file.

Alternatively, one can set `LD_LIBRARY_PATH` to whatever directory snappy is installed in before issuing `cargo build`.

If we always want to support `/usr/lib` and `/usr/local/lib`, I would suggest to go with your original code, and maybe add my proposal on top of it (at least I would benefit from it the next time i try to install a more recent snappy version to my user private directory.)